pep-0708.rst

```diff
@@ -0,0 +1,846 @@
+PEP: 708
+Title: Preventing Dependency Confusion Attacks
```

Comment:
I'm struggling to come up with a great title for this (and tbh I'm not super happy with the current title either).

By itself the changes to the repository API do not do anything to prevent dependency confusion attacks, rather they provide installers with enough information to prevent dependency confusion attacks with less collateral damage. I thought about focusing the title, abstract, etc entirely on the fact we're extending the repository API to enable cross repository namepspaces, but then the rationale or motivation section gets awkward I think, because the driver of this change is still dependency confusion.

Reply:
I've pushed a change now that doesn't change the title (yet anyways) but that refactors the Abstract, Motivation, Rationale, and Specification to try and make the above clearer. I'm still coming up with a blank for a better title though, I'll keeping noodling on it.

Reply:
Don't frown but this is a language-ish question; so I used an LLM (ChatGPT) to generate a bunch of PEP titles.

It came up with "Mitigating Dependency Confusion Attacks with Namespace Extensions" which... seems OK? If you'd like to look at the word soup it generated, here's a [Gist](https://gist.github.com/pradyunsg/777d419fec69e5be3bad9ccb77816849) of it.

Reply:
> I'm struggling to come up with a great title for this (and tbh I'm not super happy with the current title either).

Yeah, same—I tried playing around with a couple different avenues but I couldn't find any that seemed to work well without being too long (which is why I didn't actually suggest any, heh).

> It came up with "Mitigating Dependency Confusion Attacks with Namespace Extensions" which... seems OK? If you'd like to look at the word soup it generated, here's a [Gist](https://gist.github.com/pradyunsg/777d419fec69e5be3bad9ccb77816849) of it.

Riffing off that and based on what @dstufft said above, how about something like

```suggestion
Title: Extending the Repository API to Mitigate Dependency Confusion Attacks
```

Reply:
That works for me